### PR TITLE
T8 new end point GET /api/articles

### PR DESCRIPTION
**Pull request by RaymondBibby**
Happy path:
Status 200: responds with the correct body and keys, including comment count

Sad Path:
Status 400: bad request due to misspelled/ specified path

### CODE REVIEW

File: models/news_models.js [hunk 1]
```diff
@@ -28,4 +28,15 @@ exports.fetchUsers = async () => {
 exports.updateArticleById = async ( { article_id }, { inc_votes } ) => {
     const result = await db.query('UPDATE articles SET votes = votes + $1 WHERE article_id=$2 RETURNING *;', [inc_votes, article_id])
     return result.rows
+}
+
+exports.fetchArticles = () => {
+    
+    return Promise.all( [ getAllArticles(), getCommentCountAllArticles() ] )
```
Comment (k-huggs) [inline models/news_models.js]: this has been slightly over complicated, you can add a comment count with a refactor of your sql query.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -209,4 +209,47 @@ describe("GET: /api/users", () => {
...
+               articles.forEach((article) => {
+                    expect(article).toEqual(
+                        expect.objectContaining({
+                            author: expect.any(String),
+                           title: expect.any(String),
+                           article_id: expect.any(Number),
+                          topic: expect.any(String),
+                          created_at: expect.any(String),
+                          votes: expect.any(Number),
+                          comment_count: expect.any(String)
```
Comment (k-huggs) [inline __tests__/app.test.js]: comment count should be an integer not string

File: __tests__/app.test.js [hunk 3]
```diff
@@ -209,4 +209,47 @@ describe("GET: /api/users", () => {
...
+                      })
+                    )
+                })
+            })
+    })
+    it("Status 400: resposnds with a 400 bad request and an appropriate message when an invalid end point is provided, e.g., misspelling", () => {
+        return request(app)
+            .get('/api/articlles')
```
Comment (k-huggs) [inline __tests__/app.test.js]: this would be a 404 error not a 400 error

File: __tests__/app.test.js [hunk 4]
```diff
@@ -209,4 +209,47 @@ describe("GET: /api/users", () => {
...
+            .expect(400)
+            .then((  {body} )=> {
+                expect(body.msg).toBe("Invalid input, no such end point exists");
+            })
+    })
+    it("Status 400: resposnds with a 400 bad request and an appropriate message when an invalid end point is provided, e.g., /api/", () => {
+        return request(app)
+            .get('/api/users/articless/Utter_rubbish')
```
Comment (k-huggs) [inline __tests__/app.test.js]: this would be testing a different endpoint to the one specified in your describe block, also would still be a 404 not 400

**Comment by k-huggs** [comment]
Hi pal, there's a slight confusion between 404 and 400 errors on here.
You can merge once changing the errors and data type of comment count, If you need some assistance with how you can do the comment count in one sql query rather than the current way of doing it stick in an Nchelp

File: __tests__/app.test.js (hunk 2 above)
Comment (RaymondBibby) [reply __tests__/app.test.js]: I have updated all tests to account for type number with all tests passing now :)

File: __tests__/app.test.js (hunk 4 above)
Comment (RaymondBibby) [reply __tests__/app.test.js]: I have deleted test since it is fro wrong endpoint

File: __tests__/app.test.js (hunk 3 above)
Comment (RaymondBibby) [reply __tests__/app.test.js]: All test suite updated fro correct use of 400/ 404

**Pull request merged**